bob-common/src/metrics/collector/recorder.rs

```diff
@@ -1,23 +1,44 @@
 use metrics::{GaugeValue, Key, Recorder};
 use tokio::sync::mpsc::Sender;
+use std::{fmt::{Display, Formatter, Result as FMTResult}};
 
 use super::snapshot::{Metric, MetricInner};
+use crate::interval_logger::IntervalLoggerSafe;
+
+const ERROR_LOG_INTERVAL_MS: u64 = 5000;
+
+#[derive(Hash, PartialEq, Eq)]
+struct PushMetricError {
+    err: String,
+}
+
+impl Display for PushMetricError {
+    fn fmt(&self, f: &mut Formatter<'_>) -> FMTResult {
+        write!(
+            f,
+            "Can't send metric to thread, which processing metrics: {}",
+            self.err
+        )
+    }
+}
 
 pub(crate) struct MetricsRecorder {
     tx: Sender<Metric>,
+    error_logger: IntervalLoggerSafe<PushMetricError>
 }
 
 impl MetricsRecorder {
     pub(super) fn new(tx: Sender<Metric>) -> Self {
-        Self { tx }
+        Self {
+            tx,
+            error_logger: IntervalLoggerSafe::new(ERROR_LOG_INTERVAL_MS, log::Level::Error),
+        }
     }
 
     fn push_metric(&self, m: Metric) {
         if let Err(e) = self.tx.try_send(m) {
-            error!(
-                "Can't send metric to thread, which processing metrics: {}",
-                e
-            );
+            let action = PushMetricError {err: e.to_string()};
```

Comment:
I don't like this `to_string()` conversion that will happen on every metrics, when Graphite is not available. According to the documentation there are 2 types of error possible: when closed and when no capacity (https://docs.rs/tokio/0.1.22/tokio/sync/mpsc/error/struct.TrySendError.html). I suggest to convert `PushMetricError` into enum with 3 states: `Unknown`, `Closed`, `NoCapacity`. Here you can convert to that enum with functions `is_closed()`, `is_full()`